src/components/RateAgentModal/index.tsx

```diff
@@ -103,9 +104,14 @@ export default function RateAgentModal({
             ))}
         </div>
         <div className="d-grid gap-3">
-          <Button size="lg" disabled={isSubmitting} type="submit">
+          <AsyncButton
+            loading={isSubmitting}
+            size="lg"
+            disabled={isSubmitting}
```

Comment:
(Non-Blocking) Not technically needed with the new disabling logic in the component.